source/PlayerInfo.cpp

```diff
@@ -465,15 +465,23 @@ void PlayerInfo::Save() const
 		if(saved.GetDate() != date.ToString())
 		{
 			string root = filePath.substr(0, filePath.length() - 4);
-			string files[4] = {
-				root + "~~previous-3.txt",
-				root + "~~previous-2.txt",
-				root + "~~previous-1.txt",
-				filePath
-			};
-			for(int i = 0; i < 3; ++i)
-				if(Files::Exists(files[i + 1]))
-					Files::Move(files[i + 1], files[i]);
+			const int previousCount = Preferences::GetPreviousSaveCount();
+			vector<string> files;
+			files.reserve(previousCount + 1);
+			const string rootPrevious = root + "~~previous-";
+			for(int i = 0; i < previousCount; ++i)
```

Comment:
```suggestion
			for(int i = 0; i < previousCount - 1; ++i)
```
not sure, but I think this is correct? do doublecheck
but say its 2 that would mean we do it for i = 0 and i = 1, which seems correct to me

Reply:
If `previousCount` is 2, we want this loop to run twice, to insert the string ending `previous-2.txt` and then the string ending `previous-1.txt`.
If the condition is `i < previousCount - 1`, with `previousCount = 2`, it'll be `i < 1`, this means it will only run the loop body if `i < 1`, which is only satisfied by `i = 0` (since we start at 0 and add 1 at the end of each iteration).
This means it'll only run once, and insert the string ending `previous-2`, but not `previous-1`.

https://en.cppreference.com/w/cpp/language/for